jobs/haproxy/spec

```diff
@@ -128,15 +128,24 @@ properties:
           - 10.20.10.11
           backend_port: 80 # optional - sets backend port - otherwise defaults to `port`
           ssl: true        # optional - enables ssl, and uses the `ha_proxy.ssl_pem` provided key
+
   ha_proxy.resolvers:
     description: "List of DNS servers"
     example:
       resolvers:
-        private: 10.0.0.2
-        public: 8.8.8.8
+        - private: 10.0.0.2
+          public: 8.8.8.8
   ha_proxy.dns_hold:
     description: "DNS Hold time"
     default: 10s
   ha_proxy.accept_proxy:
     description: "Allow accept proxy"
     default: false
+  ha_proxy.allow_delayed_resolution:
+    description: "Allow HAProxy to start even when a DNS may not resolve yet"
+  ha_proxy.init_addr:
```

Comment:
This is overly complicated.  I recommend abstracting this to take a single argument of a either an IP address or 'none', which will be appended to the end of ["last", "libc"] internally. Absent, it should default to nil.  The use cases of having full control over this attribute is extremely limited and esoteric.  I also recommend using fallback_addr to better catch the intent of this enhancement.

```
<% 
  chain = [ "last", "libc" ]
  chain << p("ha_proxy.fallback_addr") if p("ha_proxy.fallback_addr")
%>
  default-server init-addr <%= chain.join(",") %>
```